docs/data-cleanup-mechanisms.md

```diff
@@ -0,0 +1,106 @@
+# Data cleanup mechanisms
+
+There are several mechanisms regarding the cleanup of data in EmailAlertApi.
+This doc gives an overview of each.
+
+## Nullifying email addresses
+
+Updates the `address` of old unsubscribed `Subscriber` records to `nil`. We do
+this to comply with GDPR, specifically so we're not retaining users data for
+longer than needed.
+
+We nullify subscribers in two different scenarios.
+
+Firstly, once they've been unsubscribed from all subscriptions for over 28
+days. We could nullify user's email addresses immediately after unsubscribing
+but this would limit our ability to assist in user support queries. E.g.
+restoring a users subscriptions after they've unsubscribed by accident which
+would be impossible to do if the system had nullified their address
+immediately after unsubscribing.
+
+Secondly, subscribers are nullified which are created more than 28 days ago and
+don't have any subscriptions at all (active or ended). It shouldn't be possible
+for subscribers to exist without subscriptions but these are queried and
+removed anyway to prevent storing user's data indefinitely if, for whatever
+reason, a subscriber ends up with no subscriptions. This has happened before,
+unintentionally, see [here][sub bug] and [here][sub bug two].
+
+The [NullifySubscribersWorker] runs [every hour].
+
+[every hour]: https://github.com/alphagov/email-alert-api/blob/master/config/sidekiq.yml#L26-L28
+[NullifySubscribersWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/nullify_subscribers_worker.rb
+[sub bug]: https://github.com/alphagov/email-alert-api/pull/1462/commits/053859c5962eef104256661f28727b08a43e3d31
+[sub bug two]: https://github.com/alphagov/email-alert-api/pull/1462/commits/a4b3e82801d79abd3989b1dd60ffd499e7ce82ba
+
+## Archiving emails
+
+We archive emails to S3 using the [EmailArchiveWorker]. This is done to support
+long term analytics as we can query the S3 bucket through [Athena].
+
+Emails are eligible for archiving immediately after creation as long as they
+haven't already been archived (based on `archived_at` value).
+
+The [archive worker runs every hour].
+
+[Athena]: https://github.com/alphagov/email-alert-api/blob/master/docs/analytics.md#analytics-through-athena
+[EmailArchiveWorker]: https://github.com/alphagov/email-alert-api/blob/master/app/workers/email_archive_worker.rb
+[archive worker runs every hour]: https://github.com/alphagov/email-alert-api/blob/ec075c7198b965d3771726946dcd37783b9de9c5/config/sidekiq.yml#L29-L31
+
+## Deleting emails
+
+The [EmailDeletionWorker] deletes emails after they've been archived for over
```

Comment:
```suggestion
The [EmailDeletionWorker] deletes emails after 
```

Related to the above comment about archiving being deprecated: we'll just be deleting emails 7 days after they were created.

Reply:
I've left how it works as is but have added the future suggestion to the note as the top of the section.

I'd find it strange if I looked at this documentation and it was telling me that the code did something different to what it actually does, currently.